synadm/cli/matrix.py

```diff
@@ -56,7 +56,8 @@ def login_cmd(helper, user_id, password):
         else:
             password = click.prompt("Password", hide_input=True)
 
-    login = helper.matrix_api.user_login(user_id, password)
+    mxid = helper.matrix_api.generate_mxid(user_id)
+    login = helper.matrix_api.user_login(mxid, password)
```

Comment:
We could enclose the mxid generation inline with the acutally api function call everywhere. I don't have a strong opinion on it, just an idea. I don't think if that's nicer or not, and possible to stay within the 79 characters limit, also it's quite some work to change that everywhere. You decide, it really is just an idea! :-)

```
login = helper.matrix_api.user_login(
    helper.matrix_api.generate_mxid(user_id),
    password
)
```

Reply:
For now I think that we should leave the structure as it is. The current way where the MXID generation is unraveled from the actual command is a bit more readable in my opinion. As you already mentioned it, for some commands it could be a bit tricky to stay within the 79 characters line limit. On the other hand the format you proposed seems to be more elegant and compact. I will think about your suggestion over the next days and will let you know what my final thoughts are :)